force-app/main/default/lwc/modal/__tests__/modal.test.js

```diff
@@ -108,4 +108,36 @@ describe('c-modal', () => {
                 expect(modalContainerElementHide).toBeNull();
             });
     });
+
+    it('is accessible when modal shown and public header property is set', () => {
+        const HEADER = 'The modal header';
+
+        // Create initial element
+        const element = createElement('c-modal', {
+            is: Modal
+        });
+        element.header = HEADER;
+        element.show();
+        document.body.appendChild(element);
+
+        return Promise.resolve().then(() => expect(element).toBeAccessible());
+    });
+
+    it('is accessible when modal shown and it no public header property is set', () => {
+        // Create initial element
+        const element = createElement('c-modal', {
+            is: Modal
+        });
+
+        element.show();
+        document.body.appendChild(element);
+
+        // Append element to slot so that we comply with accessibility
+        const divEl = document.createElement('div');
+        divEl.textContent = 'My Header';
+        const slotEl = element.shadowRoot.querySelector('slot[name="header"]');
+        slotEl.appendChild(divEl);
```

Comment:
this line is throwing an error when running `npm run test:unit`. I recall discussing that not setting the header property is automatically breaking accessibility rules. Was that here?

Reply:
Here's the error:

```
 PASS  force-app/main/default/lwc/modal/__tests__/modal.test.js
  ● Console

    console.error
      [LWC error]: The `appendChild` method is available only on elements that use the `lwc:dom="manual"` directive.

      137 |         divEl.textContent = 'My Header';
      138 |         const slotEl = element.shadowRoot.querySelector('slot[name="header"]');
    > 139 |         slotEl.appendChild(divEl);
          |         ^
      140 | 
      141 |         return Promise.resolve().then(() => expect(element).toBeAccessible());
      142 |     });
```

Reply:
just thinking out loud, is the fact we can't test without errors indicating that the component itself needs revisiting? there have been critiques of how this component is constructed before.

Reply:
Good catch! I didn't see the error because the test appears as passed (it's because of this https://github.com/salesforce/lwc/issues/885) despite of it being logged. 

I think we should have a way to test this test case regardless of the discussion if the component is constructed badly. I've talked to Pierre-Marie, and there's no way to do this at the moment. There's an item in the LWC backlog to allow passing slot content on component creation through createElement, but yet in backlog. I'll leave a comment about that.